directly implement the models instead of inheriting from kolibri

66eli77 opened the pull request:
not ready to merge yet, comments wanted :)

66eli77 committed:
directly implement the models instead of inheriting from kolibri

jayoshih commented on contentcuration/contentcuration/models.py:
```diff
@@ -11,8 +11,14 @@
+    name = models.CharField(max_length=200)
+    description = models.CharField(max_length=400, blank=True)
+    author = models.CharField(max_length=400, blank=True)
+    theme = models.CharField(max_length=400, blank=True)
```
Not sure what theme refers to- is it needed on the content curation side?

jayoshih commented on contentcuration/contentcuration/models.py:
```diff
@@ -43,7 +49,7 @@ class TopicTree(models.Model):
     )
     root_node = models.ForeignKey(
-        'Node',
+        'ContentMetadata',
```
When it comes to converting content curation to content app nodes, will having the same name complicate the plugin to do this?

jayoshih commented on contentcuration/contentcuration/models.py:
```diff
@@ -61,17 +67,31 @@ class Meta:
-    tag_type = models.CharField(max_length=30, null=True, blank=True)
+class ContentTag(models.Model):
+    tag_name = models.CharField(max_length=30, blank=True)
+    tag_type = models.CharField(max_length=30, blank=True)
```
Might need to make this allow for null values

jayoshih commented on contentcuration/contentcuration/models.py:
```diff
@@ -61,17 +67,31 @@ class Meta:
+    kind = models.ForeignKey('ContentKind', related_name='content_metadatas', blank=True, null=True)
+    slug = models.CharField(max_length=100)
+    total_file_size = models.IntegerField()
+    available = models.BooleanField(default=False)
```
Is this needed on the content curation side?

jayoshih commented on contentcuration/contentcuration/models.py:
```diff
@@ -11,8 +11,14 @@
+    description = models.CharField(max_length=400, blank=True)
+    author = models.CharField(max_length=400, blank=True)
+    theme = models.CharField(max_length=400, blank=True)
+    subscribed = models.BooleanField(default=False)
```
Seems like something that would be more for learners rather than curators? Or can curators subscribe to other channels?

jayoshih commented on contentcuration/contentcuration/models.py:
```diff
@@ -61,17 +67,31 @@ class Meta:
+    parent = TreeForeignKey('self', null=True, blank=True, related_name='children', db_index=True)
+    tags = models.ManyToManyField(ContentTag, symmetrical=False, related_name='tagged_content', blank=True)
+    sort_order = models.FloatField(max_length=50, default=0, verbose_name=_("sort order"), help_text=_("Ascending, lowest number shown first"))
+    license_owner = models.CharField(max_length=200, blank=True, help_text=_("Organization of person who holds the essential rights"))
```
Might help to allow this to be null for now

66eli77 committed:
add content constants

66eli77 committed:
change to new content model schema

66eli77 committed:
get rid of 'available' field

66eli77 committed:
fix the field names in serializers

66eli77 committed:
add constants for Licenses

66eli77 committed:
rename NodeModel and NodeCollection to be aligned with the new schema

66eli77 committed:
get rid of presets.json and associated models and functions
prepopulating model will be performed at the server-side

66eli77 committed:
change names on front-end to work with the new schema

66eli77 committed:
add fixture for server-side database initialization
run the following command to populate the database
manage.py loaddata initial_data --setings=contentcuration.dev_settings

66eli77 committed:
add fileformat serializer and expose fileformat_list to templates

66eli77 committed:
retrieve fileforamt_list in template

66eli77 committed:
correct collection name

66eli77 committed:
add content_copy tracking to avoid file bloating overtime.
but since everything is in one database in content curation, we can probably have something simpler to achieve the same thing
need refactorying

66eli77 committed:
add method content_copy_name help method to generate checksum for files in filefield
add ContentCopyStorage class for filefield storage

66eli77 committed:
CharField should not set null=True

66eli77 committed:
change nameing for content relationship
